biz.aQute.bndlib/test/aQute/bnd/classfile/renamer/SignatureParserTest.java

```diff
@@ -0,0 +1,133 @@
+package aQute.bnd.classfile.renamer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.Test;
+
+import com.github.curiousoddman.rgxgen.RgxGen;
+
+import aQute.bnd.classfile.renamer.ClassFileRenamer.SignatureParser;
+import aQute.bnd.classfile.renamer.ClassFileRenamer.SignatureRenamer;
+import aQute.lib.io.IO;
+
+public class SignatureParserTest {
+	final static Random						random			= new Random();
+	final static RgxGen						TYPE_GENERATOR	= new RgxGen(
+		"[[]{0,2}((I|B|C|L[abcd文efz$][ab\u0000\u0016cd0234文ef$]{2,8}(/[ab文_cde3434f$]{2,8}){0,3};)");
+	final ClassFileRenamer.SignatureParser	sp				= new ClassFileRenamer.SignatureParser();
+
+	@Test
+	public void testSignatures() throws Exception {
+		AtomicInteger i = new AtomicInteger(0);
+
+		Files.lines(IO.getFile("testresources/shade/signatures.txt")
+			.toPath())
+			.forEach(l -> {
+				sp.parse(l);
+				i.incrementAndGet();
+			});
+
+		System.out.println(i);
+	}
+
+	@Test
+	public void testSpecial() {
+		sp.parse(
+			"(I)Lorg/springframework/security/config/annotation/web/configurers/openid/OpenIDLoginConfigurer<TH;>.AttributeExchangeConfigurer.AttributeConfigurer;");
+	}
+
+	@Test
+	public void testRenamer() {
+		String s = "(I)Lorg/springframework/security/config/annotation/web/configurers/openid/OpenIDLoginConfigurer<TH;>.AttributeExchangeConfigurer.AttributeConfigurer;";
+		Map<String, String> map = new HashMap<>();
+		map.put("org/springframework/security/config/annotation/web/configurers/openid/OpenIDLoginConfigurer", "foo");
+		SignatureRenamer sr = new SignatureRenamer(map::get);
+		String rename = sr.rename(s);
+		System.out.println(rename);
```

Comment:
There are no test assertions in this method.

Reply:
ok